src/metrics/src/lgalloc.rs

```diff
@@ -0,0 +1,98 @@
+                stats: lgalloc::LgAllocStats,
+                size_class: BTreeMap<usize, LgMetricsSC>,
+                $([<$name $suffix>]: raw::UIntGaugeVec,)*
+            }
+            struct LgMetricsSC {
+                $([<$name $suffix>]: GenericGauge<AtomicU64>,)*
+            }
+            impl LgMetrics {
+                fn new(registry: &MetricsRegistry) -> Self {
+                    Self {
+                        size_class: BTreeMap::default(),
+                        stats:  lgalloc::LgAllocStats::default(),
+                        $([<$name $suffix>]: registry.register(mz_ore::metric!(
+                            name: concat!(stringify!($namespace), "_", stringify!($name), $suffix),
+                            help: $desc,
+                            var_labels: ["size_class"],
+                        )),)*
+                    }
+                }
+                fn update(&mut self) {
+                    lgalloc::lgalloc_stats(&mut self.stats);
+                    for sc in &self.stats.size_class {
+                        let sc_stats = self.size_class.entry(sc.size_class).or_insert_with(|| {
+                            LgMetricsSC {
+                                $([<$name $suffix>]: self.[<$name $suffix>].with_label_values(&[&sc.size_class.to_string()]),)*
+                            }
+                        });
+                        $(sc_stats.[<$name $suffix>].set(($conv)(u64::cast_from(sc.$name), sc));)*
+                    }
+                }
+            }
+        }
+    };
+}
+
+fn normalize_by_size_class(value: u64, stats: &SizeClassStats) -> u64 {
+    value * u64::cast_from(stats.size_class)
+}
+
+fn id(value: u64, _stats: &SizeClassStats) -> u64 {
+    value
+}
+
+metrics! {
+    mz_metrics_lgalloc
+    (allocations, "Number of region allocations in size class", "_total", id),
+    (area_total_bytes, "Number of bytes in all areas in size class", "", id),
+    (areas, "Number of areas backing size class", "_total", id),
+    (clean_regions, "Number of clean regions in size class", "_total", id),
+    (clean_regions, "Number of clean regions in size class", "_total_bytes", normalize_by_size_class),
+    (deallocations, "Number of region deallocations for size class", "_total", id),
+    (free_regions, "Number of free regions in size class", "_total", id),
+    (free_regions, "Number of free regions in size class", "_total_bytes", normalize_by_size_class),
+    (global_regions, "Number of global regions in size class", "_total", id),
+    (global_regions, "Number of global regions in size class", "_total_bytes", normalize_by_size_class),
+    (refill, "Number of area refills for size class", "_total", id),
+    (slow_path, "Number of slow path region allocations for size class", "_total", id),
+    (thread_regions, "Number of thread regions in size class", "_total", id),
+    (thread_regions, "Number of thread regions in size class", "_total_bytes", normalize_by_size_class)
```

Comment:
Are all the `_total` metrics really counting up from the beginning of process start? If they are instead reflecting the current state they shouldn't have that suffix.

Also a nit: When you have both a unit and a `_total` suffix, the unit should come first. See the examples in https://prometheus.io/docs/practices/naming/.

Reply:
They're counting from the first time lgalloc was enabled. Is this sufficient for a _total suffix?

Reply:
Yep! From the link above:
> Note that an accumulating count has `total` as a suffix, in addition to the unit if applicable.